### DISCUSSION TITLE
PropagateProxyEnv = false should remove proxy settings retrieved from os.ProxySettings

### DISCUSSION THREAD

**Discussion by pixeldrew** [lima-vm/lima#698] [Q&A] [answered]
### Description

Was banging my head against this, wondering why my local dev proxy was blowing up my vm. 

https://github.com/lima-vm/lima/blob/9e7ee971d26b565c651fc7c0359e8b42472247d9/pkg/cidata/cidata.go#L25

If I set propagateProxyEnv to false the proxy settings shouldn't be moved to the env object.

Or am I reading this wrong?

**Comment by jandubois** [comment, marked as answer]
Lima will set the proxy environment variables from your network settings in the system preferences, and the `env` key in your `lima.yaml`.

If `propagateProxyEnv` is `true`, then settings from the process environment of your `limactl start` command may override these settings. If it is `false`, then the process environment should be ignored.

**Reply by pixeldrew** [reply]
I'm ok with that, is there an option to disable proxy entirely?

**Reply by jandubois** [reply]
No, there isn't (beyond removing your proxy settings from your network preferences). You can override them from `lima.yaml`, but you can't remove them this way.

Can you explain the use case why you set global proxy settings, but don't want them to be used?

**Reply by pixeldrew** [reply]
My company laptop includes a proxy content filter that does MITM decrypting for specific websites, product name is "netskope" + "crowdstrike" it adds these proxy values to the default interface and we can't disable it. ROOT CA for this client is installed in the keyvault. Since the VM doesn't include this CA it blows up when containerd fetches containers from github's registry, I tried adding the CA to /usr/local/share/ca-certificates/netscope using openssl s_client but it doesn't connect using the proxy values.... Turns out that the proxy setting isn't used by openssl s_client so the VM can directly connect to github without using netskope. Just have to unset all the proxy env vars and containerd should be able to download images.
